view/Requirements.php

```diff
@@ -1482,20 +1515,20 @@ public function getCombinedFilesEnabled() {
 	 * any of these files have changed.
 	 *
 	 * @param array $fileList List of files
-	 * @return int Last modified timestamp of these files
+	 * @return string SHA1 bashed file hash
 	 */
-	protected function getEntropyOfFiles($fileList) {
-		// file exists, check modification date of every contained file
+	protected function hashOfFiles($fileList) {
+		// Get hash based on hash of each file
 		$base = Director::baseFolder() . '/';
-		$srcLastMod = 0;
+		$hash = '';
 		foreach($fileList as $file) {
 			if(file_exists($base . $file)) {
-				$srcLastMod = max(filemtime($base . $file), $srcLastMod);
+				$hash .= sha1_file($base . $file);
 			} else {
 				throw new InvalidArgumentException("Combined file {$file} does not exist");
```

Comment:
This doesn't account for dynamically included files, but that may no longer be the case, if so we should probably throw an exception on the include also.